apps/src/tutorialExplorer/tutorialSet.jsx

```diff
@@ -14,76 +14,79 @@ const TutorialSet = React.createClass({
-   * But if we do have a filter set for a filter group, and the tutorial is tagged
-   * for that filter group, then at least one of the active filters must match a tag.
-   * e.g. If the user chooses two platforms, then at least one of the platforms
-   * must match a platform tag on the tutorial.
-   * A similar check for language is done first.
-   * In the case that this.props.specificLocale is true, we do something slightly
-   * different.  We don't show tutorials that don't have any language tags, and we
-   * reject tutorials that don't have the current locale explicitly listed.  This
-   * allows us to return a set of tutorials that have explicit support for the
-   * current locale.
-   *
-   * @param {object} tutorial - Single tutorial, containing a variety of
-   *   strings, each of which is a list of tags separated by commas, no spaces.
-   */
-  filterFn(tutorial) {
+  statics: {
+    /**
+     * Returns true if we should show this item based on current filter settings.
+     * It goes through all active filter categories.  If no filters are set for
+     * a filter group, then that item will default to showing, so long as no other
+     * filter group prevents it from showing.
+     * But if we do have a filter set for a filter group, and the tutorial is tagged
+     * for that filter group, then at least one of the active filters must match a tag.
+     * e.g. If the user chooses two platforms, then at least one of the platforms
+     * must match a platform tag on the tutorial.
+     * A similar check for language is done first.
+     * In the case that this.props.specificLocale is true, we do something slightly
+     * different.  We don't show tutorials that don't have any language tags, and we
+     * reject tutorials that don't have the current locale explicitly listed.  This
+     * allows us to return a set of tutorials that have explicit support for the
+     * current locale.
+     *
+     * @param {object} tutorial - Single tutorial, containing a variety of
+     *   strings, each of which is a list of tags separated by commas, no spaces.
+     */
+    filterFn(tutorial) {
 
-    // First check that the tutorial language doesn't exclude it immediately.
-    // If the tags contain some languages, and we don't have a match, then
-    // hide the tutorial.
-    if (tutorial.languages_supported) {
-      const languageTags = tutorial.languages_supported.split(',');
-      const currentLocale = this.props.locale;
-      if (languageTags.length > 0 &&
-        !languageTags.includes(currentLocale) &&
-        !languageTags.includes(currentLocale.substring(0,2))) {
+      // First check that the tutorial language doesn't exclude it immediately.
+      // If the tags contain some languages, and we don't have a match, then
+      // hide the tutorial.
+      if (tutorial.languages_supported) {
+        const languageTags = tutorial.languages_supported.split(',');
+        const currentLocale = this.props.locale;
```

Comment:
Okay, I'd been misunderstanding how you were using this. It makes more sense to me now.

The way you're using filterFn it seems inaccurate to call it a static, as it ends up depending on having an instance of TutorialSet.

As I think on this though, the way you have it might be the best approach. I'll see if I think of anything better by tomorrow morning (or a good way of making it a little more clear that filterFn ends up depending on being bound to an instance of TutorialSet)